rosbag2_snapshot/src/snapshotter.cpp

```diff
@@ -985,6 +1042,20 @@ void Snapshotter::handle_accepted(const std::shared_ptr<rclcpp_action::ServerGoa
     }
   }
 
+  // Use RAII to ensure cloned buffers are cleaned up properly  
+  auto cleanup_buffers = rcpputils::make_scope_exit([&cloned_buffers, this]() {
+    // Calculate total memory before cleanup for logging  
+    size_t total_memory_freed = 0;
+    for (const auto& buffer_pair : cloned_buffers) {
+      if (buffer_pair.second) {
+        total_memory_freed += buffer_pair.second->size_;
+      }
+    }
+    RCLCPP_INFO(get_logger(), "Cleaned up rosbag cloned buffers, freed ~%.1f MB", 
```

Comment:
Do we want this log? Is not too noisy on every call of the tool or you think is fine? is not too spammy?